src/components/FormLogin/index.test.tsx

```diff
@@ -1,20 +1,62 @@
-import { ChakraProvider } from '@chakra-ui/react';
-import { render, screen, waitFor, within } from '@testing-library/react';
-import userEvent from '@testing-library/user-event';
+import { faker } from '@faker-js/faker';
+import { act, fireEvent, render, screen, within } from '@testing-library/react';
 
-import { theme } from '@/theme';
+import { useAppSelector, useAppDispatch } from '@/hooks/store';
+import { initialState as formLoginInitialState } from '@/reducers/FormLogin';
+import * as formLoginAsyncActions from '@/reducers/FormLogin/actions';
+import TestProvider from '@/tests/TestProvider';
+import { Login } from '@/types/form';
 
 import { flashToastTestIds } from '../FlashToast/index';
 import FormLogin, { formLoginTestIds } from './index';
 
+jest.mock('@/hooks/store');
+jest.mock('@/reducers/FormLogin/actions');
+jest.mock('@/adapters/Authentication');
+
+const mockDispatch = jest.fn();
+
 describe('FormLogin', () => {
   const setup = () =>
     render(
-      <ChakraProvider resetCSS theme={theme}>
+      <TestProvider>
         <FormLogin />
-      </ChakraProvider>
+      </TestProvider>
     );
 
+  const successResponse: object = {
+    data: {},
+  };
+  const validEmail = faker.internet.email();
+  const validPassword = faker.internet.password();
+  const invalidEmail = 'INVALID_MAIL';
+  const invalidPassword = 'INVALID_PASSWORD_LENGTH';
+
+  const typeUserCredential = ({ email, password }: Login) => {
+    const inputEmail = screen.getByTestId(formLoginTestIds.inputEmail);
+    const inputPassword = screen.getByTestId(formLoginTestIds.inputPassword);
+
+    fireEvent.change(inputEmail, { target: { value: email } });
+    fireEvent.change(inputPassword, { target: { value: password } });
+  };
+
+  const clickSubmitButton = () => {
+    const submit = screen.getByTestId(formLoginTestIds.submit);
+
+    fireEvent.click(submit);
```

Comment:
Should we change to use `user-event`?

Testing-library recommends using [user-event](https://testing-library.com/docs/user-event/intro) instead of [fireEvent](https://testing-library.com/docs/dom-testing-library/api-events)

> Most projects have a few use cases for fireEvent, but the majority of the time you should probably use [@testing-library/user-event](https://testing-library.com/docs/user-event/intro).

[user-event](https://github.com/testing-library/user-event) is a companion library for Testing Library that simulates user interactions by dispatching the events that would happen if the interaction took place in a browser.

Reply:
@hoangmirs I've already used that `user-event` before but it seems didn't wok for me even thought of using with `waitFor` 🥹
That's why I change back to the `fireEvent`.